MSFragger-GUI/src/com/dmtavt/fragpipe/cmd/CmdPercolator.java

```diff
@@ -133,55 +135,60 @@ public boolean configure(Component comp, Path jarFragpipe, String percolatorCmd,
     for (Entry<InputLcmsFile, List<Path>> e : pepxmlFiles.entrySet()) {
       for (Path pepxmlPath : e.getValue()) {
         final Path pepxmlDir = pepxmlPath.getParent();
-        final String nameWithoutExt = FilenameUtils.removeExtension(pepxmlPath.getFileName().toString());
-        final String basename = remove_rank_suffix(nameWithoutExt);
-        if(!basenames.add(basename))
+        //final String nameWithoutExt = FilenameUtils.removeExtension(pepxmlPath.getFileName().toString());
+        final String nameWithoutExt = PathUtils.removeExtension(pepxmlPath.getFileName().toString(), 2, 10);

```

Comment:
@chhh Why create a new `PathUtils.removeExtension()` to replace the existing `FilenameUtils.removeExtension`?

Reply:
This one has additional parameters - how many times to remove an extension (in case of files like file.raw.pep.xml) and a limit on the length of the extension, this catches cases when somebody puts a dot in the file name.
I didn't just replace it in order to replace it, there was a real life example where the original one from apache commons failed.

Reply:
This is why I forced MSFragger to always generate `<file name>.pepXML`. If you want to support `<file name>.pep.xml`, there will be a lot of places to change and test. I think many places, including the other tools used by FragPipe, assume that the extension is everything after the last dot.

Since your case will never happen if use MSFragger, I don't think it is necessary to implement this new function to make the things more complicated.